frontend/src/context/RelayerContext.tsx

```diff
@@ -1,54 +1,131 @@
 import React, { useContext, useEffect, useState } from "react";
-import { from } from "rxjs";
-import {
-  RelayerContextProviderProps,
-  RelayerContextType,
-} from "context/interfaces";
-import { Totals } from "config/interfaces/Parachain";
-import { parachains } from "config/AvailableParachain";
-import { ApiPromiseContext, SystemContext } from "context";
 import { ApiPromise } from "@polkadot/api";
-import { useProvider } from "./ProviderContext";
+import { Header } from "@polkadot/types/interfaces";
+import { from } from "rxjs";
+import { allChains } from "config/AvailableParachain";
+import { Totals, ParachainFeed, FeedTotals } from "config/interfaces/Parachain";
+import { RelayerContextProviderProps, RelayerContextType, ApiPromiseContext } from "context";
+
+function allChainFeeds(): number[] {
+  return allChains.map((chain) => chain.feedId);
+}
+
+function extractFeed(api: ApiPromise, args: any[]): ParachainFeed {
+  const id: number = api.registry.createType("u64", args[0]).toNumber();
+  const metadata = api.registry.createType("Bytes", args[2]).toHuman();
+  const { hash, number } = JSON.parse(metadata?.toString() || "{}");
+  return {
+    feedId: id,
+    hash: hash || "",
+    number: number || "",
+    size: 0,
+    count: 0,
+    subspaceHash: "",
+  };
+}
 
-async function getFeedTotals(api: ApiPromise): Promise<Totals[]> {
-  const feedsTotals: Array<Totals> = Array<Totals>();
-  for (const { feedId } of parachains) {
-    const totals = await api.query.feeds.totals(feedId);
-    feedsTotals.push(api.registry.createType("Totals", totals));
+async function getFeedTotals(api: ApiPromise): Promise<FeedTotals[]> {
```

Comment:
It would be nice to cover `getFeedTotals` and `getInitFeeds` with unit tests at some point

Reply:
there is a task in the board to cover the front end with tests, will move it to an issue to have it in the scope